hooks/playbooks/adoption_ironic_post_oc.yml

```diff
@@ -0,0 +1,124 @@
+
+        - name: Upload ironic-python-agent kernel to glance
+          environment:
+            OS_CLOUD: overcloud
+          ansible.builtin.shell:
+            cmd: >-
+              openstack image show deploy-kernel &>/dev/null || \
+                openstack image create deploy-kernel \
+                  --public \
+                  --container-format aki \
+                  --disk-format aki \
+                  --file ironic-python-agent.kernel
+          args:
+            chdir: "{{ ansible_user_dir }}/ironic-python-agent"
+
+        - name: Upload ironic-python-agent ramdisk to glance
+          environment:
+            OS_CLOUD: overcloud
+          ansible.builtin.shell:
+            cmd: >-
+              openstack image show deploy-ramdisk &>/dev/null || \
+                openstack image create deploy-ramdisk \
+                  --public \
+                  --container-format ari \
+                  --disk-format ari \
+                  --file ironic-python-agent.initramfs
+          args:
+            chdir: "{{ ansible_user_dir }}/ironic-python-agent"
+
+    - name: Create ironic provisioning network
+      delegate_to: osp-undercloud-0
+      environment:
+        OS_CLOUD: overcloud
+      block:
+        - name: Create network
+          ansible.builtin.shell:
+            cmd: >-
+              openstack network show provisioning &>/dev/null || \
+                openstack network create provisioning \
+                  --share \
+                  --provider-physical-network baremetal \
+                  --provider-network-type flat
+
+        - name: Create subnet
+          ansible.builtin.shell:
+            cmd: >-
+              openstack subnet show provisioning-subnet &>/dev/null || \
+                openstack subnet create provisioning-subnet \
+                  --network provisioning \
+                  --subnet-range 172.20.1.0/24 \
+                  --gateway 172.20.1.1 \
+                  --allocation-pool start=172.20.1.100,end=172.20.1.199
+
+    - name: Slurp ironic_nodes.yaml from controller-0
+      delegate_to: controller-0
+      register: _ironic_nodes_slurp
+      ansible.builtin.slurp:
+        src: "{{ ansible_user_dir }}/ci-framework-data/parameters/ironic_nodes.yaml"
+
+    - name: Enroll ironic nodes
+      delegate_to: osp-undercloud-0
+      block:
+        - name: Ensure directory exists
+          delegate_to: osp-undercloud-0
+          ansible.builtin.file:
+            state: directory
+            path: "{{ ansible_user_dir }}/ci-framework-data/parameters"
+
+        - name: Write ironic_nodes.yaml on osp-unercloud-o
+          ansible.builtin.copy:
+            content: "{{ _ironic_nodes_slurp.content | b64decode }}"
+            dest: "{{ ansible_user_dir }}/ci-framework-data/parameters/ironic_nodes.yaml"
+
+        - name: Run baremetal create command to enroll the nodes in the Ironic service
+          environment:
+            OS_CLOUD: overcloud
```

Comment:
Same as for previous hard-coded OS_CLOUD

Reply:
Done